refactor: DSearch implementation

hanszhoou opened the pull request:
## Description
https://mask.atlassian.net/browse/MF-2691
https://mask.atlassian.net/browse/MF-2815
https://mask.atlassian.net/browse/MF-2830

<!-- Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change. -->

Closes # (NO_ISSUE)

## Type of change

<!-- Please remove options that are not relevant. -->

- [ ] Documentation
- [x] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Previews

<!-- Please attach screenshots if there are any visual changes. -->

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. <!-- If you don't have permission to modify the server, please let us know it. -->
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

guanbinrui committed:
refactor: renaming folder

guanbinrui committed:
chore: add DSearch API

guanbinrui committed:
chore: move types

guanbinrui committed:
refactor: renaming

guanbinrui committed:
fix: lint errors

hanszhoou committed:
chore: wip

hanszhoou committed:
chore: wip

hanszhoou committed:
chore: wip

hanszhoou committed:
chore: wip

hanszhoou committed:
chore: wip

hanszhoou committed:
chore: wip

hanszhoou committed:
Merge branch 'develop' into refactor/d-search-implementation

hanszhoou committed:
chore: type error

hanszhoou committed:
chore: wip

hanszhoou committed:
chore: wip

hanszhoou committed:
chore: remove dSearch plugin

hanszhoou committed:
refactor: trending part

zhouhanseng commented:
My VPN is too unstable to do `pnpm i` to update the pnpm yaml. I will try it tomorrow.

hanszhoou committed:
chore: pnpm

hanszhoou committed:
chore: little fix

guanbinrui commented on packages/web3-state/src/NameService.ts:
```diff
@@ -56,7 +56,7 @@ export class NameServiceState<
     }
 
-    async lookup(chainId: ChainId, name: string) {
+    lookup = async (chainId: ChainId, name: string) => {
```
What's the difference from the previous version?

hanszhoou replied:
The scope of `this`.

```
const lookup = NameService.lookup

lookup() // incorrect
```

guanbinrui committed:
refactor: DSearch implementation improvement (#8114)

* refactor: TagType

* refactor: code stlye

* refactor: adjust type

guanbinrui commented on packages/web3-providers/src/d-search/index.ts:
```diff
@@ -1,19 +1,93 @@
+export class DSearchAPI<ChainId = Web3Helper.ChainIdAll> implements DSearchBaseAPI.Provider<ChainId, NetworkPluginID> {
+    async search(
+        keyword: string,
+        options: {
```
Remove `options` and import `@masknet/web3-shared-*` stead.

guanbinrui commented on packages/web3-shared/base/src/specs/index.ts:
```diff
@@ -94,28 +94,23 @@ export enum SourceType {
-    NonFungibleToken = 'NonFungibleToken',
-    // e.g., $PUNK #PUNK
-    NonFungibleCollection = 'NonFungibleCollection',
+    TrendingTokenByKeyword = 'TrendingTokenByKeyword',
+    // e.g., 0x69af81e73a73b40adf4f3d4223cd9b1ece623074 (Mask Network)
+    TrendingTokenByAddress = 'TrendingTokenByAddress',
```
They are the same thing `FungibleToken`. And please don't remove `NonFungibleToken`. It's used in the next feature.

hanszhoou committed:
Merge branch 'develop' into refactor/d-search-implementation

hanszhoou committed:
chore: import helper func directly

hanszhoou committed:
fix: type error

hanszhoou committed:
Merge branch 'develop' into refactor/d-search-implementation

hanszhoou committed:
chore: fungible token search type

hanszhoou committed:
chore: domain

hanszhoou committed:
Merge branch 'develop' into refactor/d-search-implementation

guanbinrui committed:
feat: impl 4337 RPC methods (#8051)

* feat: impl 4337 rpc

* chore: wip

* fix: import wrong variable

* chore: add AccountTransaction

* fix: typo

* refactor: code style

* chore: impl UserTransaction

* chore: impl WalletContract

* fix: typo & ci errors

* chore: add SC Deployer

* refactor: multicall as provider

* fix: build errors

* fix: test Create2Factory

Jack-Works committed:
docs(i18n): sync i18n files from Crowdin (#8127)

* New translations en-US.json (Japanese)
docs(i18n): sync translation en-US.json for Japanese

* New translations en-US.json (Chinese Traditional)
docs(i18n): sync translation en-US.json for Chinese Traditional

* New translations en-US.json (Quenya)
docs(i18n): sync translation en-US.json for Quenya

Jack-Works committed:
fix: plugin infra exports (#8126)

* fix: plugin-infra exports

* chore: move

* fix: plugin-infra exports

* fix: linter

* fix: build error

hanszhoou committed:
Merge branch 'develop' into refactor/d-search-implementation

hanszhoou committed:
chore: cspell

hanszhoou committed:
Revert "chore: cspell"

This reverts commit 1d4ad708c53b4d6b7b5d0a50e16356b63416121c.

hanszhoou committed:
Merge branch 'develop' into refactor/d-search-implementation

hanszhoou committed:
chore: remove useless code

hanszhoou committed:
fix: mf-2815 ui

hanszhoou committed:
chore: mf-2830 add trending footer

hanszhoou committed:
chore: ui

hanszhoou committed:
Merge branch 'develop' into refactor/d-search-implementation

nuanyang233 committed:
feat: go plus approval api (#8128)

* feat: impl fungiable token approval api

* feat: impl nonfungiable approval api

* chore: cspell

* chore: ci build

guanbinrui committed:
feat: add AbstractAccountProvider (#8140)

* feat: add AbstractAccountAPI

* feat: add ContractAccountAPI

* chore: getAccountsOfOwner

* fix: lint errors

* fix: typo

* fix: bundler api

* fix: lint errors

Jack-Works committed:
refactor: move ValueRef from holowflows-kit to mask (#8141)

* chore: move ValueRef from holowflows-kit to mask

* fix: lint

* prettier

hanszhoou committed:
chore: pnpm lock

hanszhoou committed:
Merge branch 'develop' into refactor/d-search-implementation

hanszhoou committed:
chore: pnpm lock

hanszhoou committed:
chore: locale

hanszhoou committed:
Merge branch 'develop' into refactor/d-search-implementation

hanszhoou committed:
chore: locale

hanszhoou committed:
chore: pnpm lock

hanszhoou committed:
Merge branch 'develop' into refactor/d-search-implementation

hanszhoou committed:
fix: types

hanszhoou committed:
chore: pnpm

hanszhoou committed:
refactor: invoke DSearch api

hanszhoou committed:
Merge branch 'develop' into refactor/d-search-implementation

hanszhoou committed:
Merge branch 'develop' into refactor/d-search-implementation

hanszhoou committed:
fix: type error

hanszhoou committed:
fix: test

hanszhoou committed:
chore: wip

hanszhoou committed:
chore: wip

hanszhoou committed:
chore: add rules for nft

hanszhoou committed:
chore: data source list

hanszhoou committed:
chore: wip

hanszhoou committed:
Merge branch 'develop' into refactor/d-search-implementation

hanszhoou committed:
chore: pnpm

hanszhoou committed:
chore: type

hanszhoou committed:
chore: text

hanszhoou committed:
chore: token menu

hanszhoou committed:
chore: powered by

hanszhoou committed:
chore: trending popper

hanszhoou committed:
chore: trending popper

hanszhoou committed:
chore: rm debug

hanszhoou committed:
chore: remove outdated code

hanszhoou committed:
chore: remove unused code

hanszhoou committed:
chore: remove useless code

hanszhoou committed:
chore: remove useless code

hanszhoou committed:
chore: coin menu ui style

hanszhoou committed:
chore: rm debug

hanszhoou committed:
chore: rm useless code

hanszhoou committed:
refactor: remove type DataProvider

hanszhoou committed:
chore: remove blacklist since we have a rank list

hanszhoou committed:
chore: little fix

hanszhoou committed:
fix: test

hanszhoou committed:
chore: remove useless props

hanszhoou committed:
chore: not only evm bsc matic

hanszhoou committed:
chore: brief type

hanszhoou committed:
chore: add CA search result

hanszhoou committed:
chore: add deps

hanszhoou committed:
chore: trigger ci

hanszhoou committed:
Merge branch 'develop' into refactor/d-search-implementation

hanszhoou committed:
chore: simplify type

guanbinrui commented on packages/web3-shared/base/src/specs/index.ts:
```diff
@@ -604,17 +609,31 @@ export interface NonFungibleTokenSecurity {}
 
-export interface AddressResult<ChainId> extends Result<ChainId> {
-    type: SearchResultType.Address
+export interface CAResult<ChainId> extends Result<ChainId> {
```
This `CAResult` duplicate with `FungibleTokenResult` or `NonFungibleTokenResult`.

guanbinrui replied:
The `DSearchAPI` should take responsibility for returning what a contract address belongs to. A rough `CAResult` is meanless for UI.

guanbinrui commented on packages/web3-providers/src/types/Trending.ts:
```diff
@@ -187,7 +187,7 @@ export namespace TrendingAPI {
         getCoinsByKeyword: (chainId: ChainId, keyword: string) => Promise<Coin[]>
         /** Get coin info by address. */
-        getCoinInfoByAddress: (chainId: ChainId, address: string) => Promise<CoinInfo | undefined>
+        getCoinInfoByAddress: (chainId: ChainId | undefined, address: string) => Promise<CoinInfo | undefined>
```
Why accept an `undefined`?

guanbinrui commented on packages/web3-providers/tests/DSearch/index.ts:
```diff
@@ -1,12 +1,11 @@
     test('should return by sysmbol from specific list', async () => {
         const DSearch = new DSearchAPI()
-        const result = await DSearch.search('token:eth')
+        const result = await DSearch.searchToken('token:eth')
```
Why? The `token:` prefix already indicates it is searching for a token result.

guanbinrui commented on packages/web3-providers/src/NFTScan/apis/TrendingAPI.ts:
```diff
@@ -111,23 +110,26 @@ export class NFTScanTrendingAPI implements TrendingAPI.Provider<ChainId> {
-            undefined,
-        )
-        if (!result) {
+        console.log({ chainId_ })
```
```suggestion
```

guanbinrui commented on packages/web3-providers/src/DSearch/index.ts:
```diff
@@ -90,7 +118,7 @@ export class DSearchAPI<ChainId = Web3Helper.ChainIdAll, SchemaType = Web3Helper
      *
      */
-    async search(keyword: string): Promise<Array<SearchResult<ChainId, SchemaType>>> {
+    async searchToken(keyword: string): Promise<Array<SearchResult<ChainId, SchemaType>>> {
```
We search for more than tokens. The new name looks not correct to me.

hanszhoou replied:
`searchToken()` invoked by `search()`

guanbinrui commented on packages/plugins/RSS3/src/SNSAdaptor/index.tsx:
```diff
@@ -34,24 +36,74 @@ const createActivitiesTabConfig = (label: string, props: FeedPageProps, priority
 
-const ActivitiesTabConfig: Plugin.SNSAdaptor.ProfileTab = createActivitiesTabConfig('Activities', {})
-const ActivitiesTabConfigInProfileCard: Plugin.SNSAdaptor.ProfileTab = createActivitiesTabConfig('Activities', {}, 2)
+const createSearchTabConfig = (
```
CC @UncleBill, Please take a look.

guanbinrui commented on packages/web3-shared/base/src/specs/index.ts:
```diff
@@ -95,28 +95,33 @@ export enum SourceType {
-}
-
-/** @deprecated use SearchSourceType stead */
+/** @deprecated use SearchResultType stead */
```
Can we remove it now?

hanszhoou replied:
yep

guanbinrui commented on packages/mask/src/extension/popups/pages/Wallet/ConnectWallet/index.tsx:
```diff
@@ -78,7 +78,7 @@ const ConnectWalletPage = memo(() => {
         ) => {
             if (provider.type === ProviderType.MaskWallet) {
-                const connection = await Connection?.getConnection?.()
+                const connection = Connection?.getConnection?.()
```
Missed one. I will do it in another PR.

guanbinrui commented on packages/plugins/EVM/src/state/Connection/connection.ts:
```diff
@@ -154,10 +154,10 @@ class Connection implements EVM_Connection {
     }
 
-    private getOptions(
+    private getOptions = (
```
It's poor maintainability. How can others know whether if need to use the arrow function? Why not use `bind`?

hanszhoou committed:
chore: reply code review

hanszhoou committed:
chore: remove useless code

hanszhoou committed:
chore: remove debug

hanszhoou committed:
chore: testing

hanszhoou committed:
chore: move search out of trending UI

hanszhoou committed:
chore: chainId type exclude undefined

hanszhoou committed:
chore: add testing

hanszhoou committed:
chore: simplify Result type

hanszhoou committed:
chore: testing

guanbinrui commented on packages/web3-shared/base/src/specs/index.ts:
```diff
@@ -95,28 +95,27 @@ export enum SourceType {
+    // e.g., 0x6a7122B831c2B79c508A978f73f2ee23171279B3
+    EOA = 'EOA',
+    // Contract Address, not sure is fungible or nonFungible token
+    CA='CA',
```
Remove it.

hanszhoou committed:
chore: remove useless code

guanbinrui commented on packages/web3-shared/base/src/specs/index.ts:
```diff
@@ -95,28 +95,25 @@ export enum SourceType {
+    Unknown = 'Unknown'
+}
+
+export enum SearchFungibleTokenResultSubType {
```
Remove it.

guanbinrui commented on packages/web3-providers/src/NFTScan/apis/TrendingAPI.ts:
```diff
@@ -111,23 +110,25 @@ export class NFTScanTrendingAPI implements TrendingAPI.Provider<ChainId> {
-        if (!result) {
+        const result = chainId_
+            ? { collection: await this.getCollection(chainId_, id), chainId: chainId_ }
+            : await attemptUntil(
```
This API should serve to query one chain id each time.

guanbinrui commented on packages/web3-providers/src/DSearch/index.ts:
```diff
@@ -90,11 +118,37 @@ export class DSearchAPI<ChainId = Web3Helper.ChainIdAll, SchemaType = Web3Helper
      *
      */
-    async search(keyword: string): Promise<Array<SearchResult<ChainId, SchemaType>>> {
+    async searchToken(keyword: string): Promise<Array<SearchResult<ChainId, SchemaType>>> {
```
```suggestion
    private async searchToken(keyword: string): Promise<Array<SearchResult<ChainId, SchemaType>>> {
```

hanszhoou committed:
chore: move logic

hanszhoou committed:
chore: private method

hanszhoou committed:
Merge branch 'develop' into refactor/d-search-implementation

hanszhoou committed:
chore: remove UnknownResult

hanszhoou committed:
fix: breaking test

guanbinrui committed:
refactor: DSearch implementation improve (#8290)

* refactor: code style

* refactor: code style

guanbinrui committed:
refactor: code style

hanszhoou committed:
fix: search nft and token both

guanbinrui committed:
refactor: DSearch implementation improve (#8292)

* refactor: code style

* fix: typo

hanszhoou committed:
fix: collection should be search

hanszhoou committed:
fix: type

guanbinrui committed:
Refactor/d search implementation improve (#8293)

* refactor: code style

* fix: typo

* fix: lint errors
